File: Day02_TodoList/Python/todo_list.py
tasks = []

def add_task():
    task = input("Enter a new task: ")
    tasks.append({"task": task, "done": False})
    print("Task added!")

def view_tasks():
    if not tasks:
        print("No tasks found.")
        return
    for i, t in enumerate(tasks, 1):
        status = "✓" if t["done"] else "✗"
        print(f"{i}. [{status}] {t['task']}")

def mark_done():
    view_tasks()
    try:
        index = int(input("Enter task number to mark as done: ")) - 1
        tasks[index]["done"] = True
        print("Task marked as done.")
    except (IndexError, ValueError):
        print("Invalid task number.")

def delete_task():
    view_tasks()
    try:
        index = int(input("Enter task number to delete: ")) - 1
        tasks.pop(index)
        print("Task deleted.")
    except (IndexError, ValueError):
        print("Invalid task number.")

def menu():
    while True:
        print("\n1. Add Task\n2. View Tasks\n3. Mark as Done\n4. Delete Task\n5. Exit")
        choice = input("Choose an option: ")
        if choice == "1":
            add_task()
        elif choice == "2":
            view_tasks()
        elif choice == "3":
            mark_done()
        elif choice == "4":
            delete_task()
        elif choice == "5":
            break
        else:
            print("Invalid choice!")

if __name__ == "__main__":
    menu()

File: Day02_TodoList/Java/TodoList.java
import java.util.ArrayList;
import java.util.Scanner;

class Task {
    String name;
    boolean done;

    Task(String name) {
        this.name = name;
        this.done = false;
    }
}

public class TodoList {
    static ArrayList<Task> tasks = new ArrayList<>();
    static Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        while (true) {
            System.out.println("\n1. Add Task\n2. View Tasks\n3. Mark as Done\n4. Delete Task\n5. Exit");
            String choice = scanner.nextLine();

            switch (choice) {
                case "1": addTask(); break;
                case "2": viewTasks(); break;
                case "3": markDone(); break;
                case "4": deleteTask(); break;
                case "5": return;
                default: System.out.println("Invalid choice!");
            }
        }
    }

    static void addTask() {
        System.out.print("Enter a new task: ");
        String taskName = scanner.nextLine();
        tasks.add(new Task(taskName));
        System.out.println("Task added!");
    }

    static void viewTasks() {
        if (tasks.isEmpty()) {
            System.out.println("No tasks found.");
            return;
        }
        for (int i = 0; i < tasks.size(); i++) {
            String status = tasks.get(i).done ? "✓" : "✗";
            System.out.println((i + 1) + ". [" + status + "] " + tasks.get(i).name);
        }
    }

    static void markDone() {
        viewTasks();
        System.out.print("Enter task number to mark as done: ");
        try {
            int index = Integer.parseInt(scanner.nextLine()) - 1;
            tasks.get(index).done = true;
            System.out.println("Task marked as done.");
        } catch (Exception e) {
            System.out.println("Invalid task number.");
        }
    }

    static void deleteTask() {
        viewTasks();
        System.out.print("Enter task number to delete: ");
        try {
            int index = Integer.parseInt(scanner.nextLine()) - 1;
            tasks.remove(index);
            System.out.println("Task deleted.");
        } catch (Exception e) {
            System.out.println("Invalid task number.");
        }
    }
}
